test/pylib/scylla_cluster.py

```diff
@@ -113,6 +113,61 @@ def make_scylla_conf(workdir: pathlib.Path, host_addr: str, seed_addrs: List[str
     '--abort-on-ebadf', '1'
 ]
 
+# [--smp, 1], [--smp, 2] -> [--smp, 2]
+# [--smp, 1], [--smp] -> [--smp]
+# [--smp, 1], [--smp, __missing__] -> [--smp]
+# [--smp, 1], [--smp, __remove__] -> []
+# [--smp=1], [--smp=2] -> [--smp, 2]
+# [--smp=1], [--smp=__remove__] -> []
+# [--overprovisioned, --smp=1, --abort-on-ebadf], [--smp=2] -> [--overprovisioned, --smp=2, --abort-on-ebadf]
+# [], [--experimental-features, raft, --experimental-features, broadcast-tables] ->
+# [--experimental-features, raft, --experimental-features, broadcast-tables]
+def merge_cmdline_options(base: List[str], override: List[str]) -> List[str]:
+    if len(override) == 0:
+        return base
+
+    def to_dict(args: List[str]) -> Dict[str, List[str]]:
+        result: Dict[str, List[str]] = {}
+        i = 0
+        while i < len(args):
+            name = args[i]
+            if not name.startswith('-'):
+                raise ValueError(f'invalid argument name {name}, all args {args}')
+            if '=' in name:
+                name, _, value = name.partition('=')
+                i += 1
+            elif i < len(args) - 1 and not args[i + 1].startswith('-'):
+                value = args[i + 1]
+                i += 2
+            else:
+                value = None
+                i += 1
+            result.setdefault(name, []).append(value)
```

Comment:
There is little value in playing a compiler role in the PR comments.

Reply:
https://mypy-lang.org/

`pip install mypy`
then
`mypy <path-to-file>`

I recommend configuring your editor to use mypy in the background (I use vim + ALE and it works flawlessly)

Reply:
I will eventually try to setup this in CI... Although @nyh might have a different opinion :P

Reply:
But it should be possible to enforce `mypy` just in the places where we write type hints, it should ignore code without any type hints written. So it will still be optional, if someone doesn't want to use it won't be affected.